src/api/forms/service.js

```diff
@@ -84,8 +87,23 @@ export async function getForm(formId) {
  * @param {string} formId - the ID of the form
  * @throws {FailedToReadFormError} - if the file does not exist or is empty
  */
-export function getFormDefinition(formId) {
-  return formDefinition.get(formId)
+export function getDraftFormDefinition(formId) {
+  return draftFormDefinition.get(formId)
+}
+
+/**
+ *
+ * @param {string} formId - ID of the form
+ * @param {FormDefinition} formDefinition - full JSON form definition
+ */
+export async function updateDraftFormDefinition(formId, formDefinition) {
+  if (!(await formMetadata.exists(formId))) {
```

Comment:
Shall we take @davidjamesstone's suggestion here?

Then can probably remove the metadata `exists()` method again

```suggestion
  const form = await getForm(formId)

  if (!form) {
```